src/tag.cpp

```diff
@@ -57,6 +57,15 @@ HSTag::HSTag(string name_, TagManager* tags, Settings* settings)
     floating_focused.setValidator([this](bool v) {
         return this->floatingLayerCanBeFocused(v);
     });
+
+    floating.setDoc("if the entire tag is set to floating mode");
```

Comment:
For me it feels weird to extend the code by a new class in order to extend documentation/generated JSON.

Reply:
In the end, I'd like to have a command `help` with the following behaviour:
```
$ hc help tags.0.floating
type: boolean setting
current value: false
meaning: if the entire tag is set to floating mode
```
https://github.com/herbstluftwm/herbstluftwm/pull/998

Reply:
My initial dislike was directed to the latter, but seeing the example of the help command, it makes much more sense to me now. I didn't realize the potential of that plan in the commit message of other PR.

I was just confused because normally "documentation" of this sort was kind of generated by doxygen or doxygen comments, but I know that this is not something we can use here.

So I'm gonna approve this (after a few minor typos are fixed) and see where this goes :)